### PR TITLE
Update eng-common-workflow-enforcer

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Fix error in eng-common-workflow-enforcer

### CODE REVIEW

File: eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 steps:
   - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
     - pwsh: |
+        $PrSourceBranch = "$(System.PullRequest.SourceBranch)"
         $filesInCommonDir = git diff "origin/$(System.PullRequest.TargetBranch)" HEAD --name-only -- 'eng/common/*'
 
         if ((-not $PrSourceBranch.StartsWith("sync-eng/common")) -and ($filesInCommonDir.Count -gt 0))
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml:7]:
```suggestion
        if ((!"$(System.PullRequest.SourceBranch)".StartsWith("sync-eng/common")) -and ($lasteexitcode -eq 0 -and $filesInCommonDir.Count -gt 0))
```

File: eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 steps:
   - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
     - pwsh: |
+        $PrSourceBranch = "$(System.PullRequest.SourceBranch)"
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml:4]: Lets just eliminate this variable as it isn't really needed. I gave a suggest below.

**Review by weshaggard — APPROVED** [review]
Looks good but lets please test it out before merging.

**Comment by chidozieononiwu** [comment]
> Looks good but lets please test it out before merging.

I'll test it while calling it in the language repos

File: eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@ steps:
     - pwsh: |
         $filesInCommonDir = git diff "origin/$(System.PullRequest.TargetBranch)" HEAD --name-only -- 'eng/common/*'
 
-        if ((-not $PrSourceBranch.StartsWith("sync-eng/common")) -and ($filesInCommonDir.Count -gt 0))
+        if ((!"$(System.PullRequest.SourceBranch)".StartsWith("sync-eng/common")) -and ($LASTEXITCODE -eq 0) -and ($filesInCommonDir.Count -gt 0))
```
Comment (benbp) [inline eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml:6]: Will $LASTEXITCODE ever be hit? Wouldn't this script halt in devops if there's a non-zero exit code?

Comment (weshaggard) [reply eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml:6]: `git diff` should set a lastexitcode but you are correct that maybe the error preferences devops uses might cause it to fail at that point. We'd have to test to know for sure but in either case I don't think it hurts to have the check.

**Pull request merged**